src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs

```diff
@@ -150,6 +150,13 @@ internal HResult BackupAndDeleteGlobalJson(string solutionDirectory, string back
                 try
                 {
                     _fileSystem.RemoveFile(globalJson);
+
+                    var remover = new GlobalJsonRemover(_serviceProvider);
+                    HResult hr = solution.AdviseSolutionEvents(remover, out uint cookie);
```

Comment:
We should probably add foreground asserts in our code base for making this more explicit. Something for future..